beefy-primitives/src/commitment.rs

```diff
@@ -157,7 +282,7 @@ mod tests {
 		assert_eq!(
 			encoded,
 			hex_literal::hex!(
-				"3048656c6c6f20576f726c6421050000000000000000000000000000000000000000000000100000011001020304011005060708"
+				"3048656c6c6f20576f726c64210500000000000000000000000000000000000000000000000403040000000810010203041005060708"
```

Comment:
Hmm, this one is actually longer, but I think it's because we only have 4 signatures.

Do you mind adding a test with 1024 signatures and `(1024 * 2 + 2) / 3` of them being set to `Some`?

Reply:
1024? woah 😂 
Let me see, might take some time for me to get to this, is that fine?

Reply:
Just use a loop :) The signature content doesn't really matter - can be the same for each.

Reply:
Yeah, can you check this [without compression](https://github.com/paritytech/grandpa-bridge-gadget/commit/0d607aaff7da34120c1e7badaf8b4d987fdec101) and [with compression](https://github.com/paritytech/grandpa-bridge-gadget/pull/186/commits/d5d4b8850d6ae15d0c99065b7b263005e49c772a)

Reply:
Cool, afaict it's 8966 characters vs 7186 so around 20% optimisation, nice!